test/AccessLock.t.sol

```diff
@@ -0,0 +1,726 @@
+            )
+        );
+        deployCodeTo("AccessLockHook.sol:AccessLockHook", abi.encode(manager), accessLockAddress);
+        accessLockHook = AccessLockHook(accessLockAddress);
+
+        (key,) = initPool(
+            currency0, currency1, IHooks(address(accessLockHook)), Constants.FEE_MEDIUM, SQRT_RATIO_1_1, ZERO_BYTES
+        );
+
+        // Create AccessLockHook2.
+        address accessLockAddress2 = address(uint160(Hooks.ACCESS_LOCK_FLAG | Hooks.BEFORE_MODIFY_POSITION_FLAG));
+        deployCodeTo("AccessLockHook.sol:AccessLockHook2", abi.encode(manager), accessLockAddress2);
+        accessLockHook2 = AccessLockHook2(accessLockAddress2);
+
+        // Create AccessLockHook3.
+        address accessLockAddress3 = address(
+            (uint160(makeAddr("hook3")) << 10) >> 10
+                | (uint160(Hooks.ACCESS_LOCK_FLAG | Hooks.BEFORE_MODIFY_POSITION_FLAG))
+        );
+        deployCodeTo("AccessLockHook.sol:AccessLockHook3", abi.encode(manager), accessLockAddress3);
+        accessLockHook3 = AccessLockHook3(accessLockAddress3);
+
+        // Create NoAccessLockHook.
+        address noAccessLockHookAddress = address(uint160(Hooks.BEFORE_MODIFY_POSITION_FLAG));
+        deployCodeTo("AccessLockHook.sol:AccessLockHook", abi.encode(manager), noAccessLockHookAddress);
+        noAccessLockHook = AccessLockHook(noAccessLockHookAddress);
+
+        // Create AccessLockHook with NoOp.
+        address accessLockHook4Address = address(
+            uint160(
+                Hooks.NO_OP_FLAG | Hooks.ACCESS_LOCK_FLAG | Hooks.BEFORE_INITIALIZE_FLAG | Hooks.BEFORE_SWAP_FLAG
+                    | Hooks.BEFORE_MODIFY_POSITION_FLAG | Hooks.BEFORE_DONATE_FLAG
+            )
+        );
+        deployCodeTo("AccessLockHook.sol:AccessLockHook", abi.encode(manager), accessLockHook4Address);
+        accessLockHook4 = AccessLockHook(accessLockHook4Address);
+    }
+
+    function test_onlyByLocker_revertsForNoAccessLockPool() public {
+        (PoolKey memory keyWithoutAccessLockFlag,) =
+            initPool(currency0, currency1, IHooks(noAccessLockHook), Constants.FEE_MEDIUM, SQRT_RATIO_1_1, ZERO_BYTES);
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                IPoolManager.LockedBy.selector, address(modifyPositionRouter), address(noAccessLockHook)
+            )
+        );
+        modifyPositionRouter.modifyPosition(
+            keyWithoutAccessLockFlag,
+            IPoolManager.ModifyPositionParams({tickLower: 0, tickUpper: 60, liquidityDelta: 0}),
+            abi.encode(10, AccessLockHook.LockAction.Mint) // attempts a mint action that should revert
+        );
+    }
+    /**
+     *
+     *
+     * The following test suite tests that appropriate hooks can call
+     *  every function gated by the `onlyByLocker` modifier.
+     *  We call these "LockActions".
+     *  LockActions:
+     *  - Mint
+     *  - Take
+     *  - Swap
+     *  - ModifyPosition
+     *  - Donate
```

Comment:
why is Burn / Settle not tested? Is it because they are purely returning money to core?

Reply:
Ah yeah I can add tests for those